mmaction/datasets/ssn_dataset.py

```diff
@@ -0,0 +1,660 @@
+import copy
+import math
+import os.path as osp
+
+import mmcv
+import numpy as np
+from torch.nn.modules.utils import _pair
+
+from mmaction.localization.proposal_utils import temporal_iou
+from .base import BaseDataset
+from .registry import DATASETS
+from .utils import (load_localize_proposal_file, parse_directory,
+                    process_localize_proposal_list)
+
+
+class SSNInstance(object):
+    """Proposal instance of SSN.
+
+    Args:
+        start_frame (int): Index of the proposal's start frame.
+        end_frame (int): Index of the proposal's end frame.
+        num_frames (int): Total frames of the video.
+        label (int): Classification of the proposal.
+        best_iou (int): The highest IOU with the groundtruth instance.
+        overlap_self (int): Percent of the proposal's own span contained
+            in a groundtruth instance.
+    """
+
+    def __init__(self,
+                 start_frame,
+                 end_frame,
+                 num_frames,
+                 label=None,
+                 best_iou=None,
+                 overlap_self=None):
+        self.start_frame = start_frame
+        self.end_frame = min(end_frame, num_frames)
+        self.num_frames = num_frames
+        self.label = label if label is not None else -1
+        self.coverage = (end_frame - start_frame) / num_frames
+        self.best_iou = best_iou
+        self.overlap_self = overlap_self
+        self.loc_reg = None
+        self.size_reg = None
+        self.regression_targets = [0., 0.]
+
+    def compute_regression_targets(self, gt_list, fg_thresh):
+        """Compute regression targets of positive proposals.
+
+        Args:
+            gt_list (list): The list of groundtruth instances.
+            fg_thresh (float): Minimum threshold of overlap of
+                positive/foreground proposals and groundtruths.
+        """
+        # Background proposals do not have regression targets.
+        if self.best_iou < fg_thresh:
+            return
+
+        # Find the groundtruth instance with the highest IOU.
+        ious = []
+        for gt in gt_list:
+            ious.append(
```

Comment:
`ious = [temporal_iou(self.start_frame, self.end_frame, gt.start_frame, gt.end_frame)) for gt in gt_list]`